### DISCUSSION TITLE
Does `vercel dev` respect the routes config in `vercel.json`?

### DISCUSSION THREAD

**Discussion by ctrlplusb** [vercel/vercel#4481] [Help]
I tried to create a basic example utilising Create React App and a few basic routes, but as far as I can tell the `routes` configuration isn't being considered at all.

I can create a reproduction if this is expected to work.

**Comment by ctrlplusb** [comment]
Do you know what would be a dream, would be a visualiser of some sorts. JSON editor on left panel containing an example `vercel.json`, right panel an example project structure (with drop downs to toggle through project types).

Then bottom panel would allow you to type in a request path and type (e.g. GET /api/foo), which would subsequently generate a a visualisation  below showing the matched route with an arrow pointing to the function/file it would resolve to.

I feel like it's so trial and error with the `vercel.json` having a tool like this would be invaluable for quickly understanding the semantics around routing configuration.

I understand that Zero Config is by far the preferred approach, but as projects scale in various ways the routes configuration becomes helpful in many ways.

**Comment by ctrlplusb** [comment]
Hmm, it appears that the routes configuration is partially working. I note different behaviour when deploying to Vercel platform vs running locally.

**Comment by leo** [comment]
@ctrlplusb Would you mind sharing the difference in behavior that you're encountering? Like that, we can make sure the issue is resolved and a regression test is added.
